### PR TITLE
Redis support

### LINKED ISSUE #79: Redis support
## Is your feature request related to a problem? Please describe

Running Consul is a complex process that involves managing a cluster. 
A simpler solution is to have Redis support which is very easy to operate and we have also AWS support for it.

## Describe the solution

- Introduce a Redis getter and watcher and a new struct tag `redis`.
- Add builder chain methods for setting up Redis

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

Closes #79.

## Short description of the changes

- Introduce a getter
- Introduce a watcher
- Introduce new struct tag
- Add new methods to the builder
- Improved test coverage

### CODE REVIEW

**Review by c0nstantx — COMMENTED** [review]
Nice addition.
Can you squash commits with same name, please ?

**Comment by mantzas** [comment]
> Nice addition.
> Can you squash commits with same name, please ?

Is this blocking your CR? The end result should matter in the end everything will be squashed and merged to master in one commit!

**Comment by c0nstantx** [comment]
> Is this blocking your CR? The end result should matter in the end everything will be squashed and merged to master in one commit!

No. It's not blocking. It's the only comment I had.
I didn't know that we have a "Squash and Merge" policy in this repo. Could you please add it to the contribution guidelines ?

Personally I prefer "Rebase and Merge" for open source projects so that the commit log on `master` branch is more clear.

Thanks.

**Review by c0nstantx — APPROVED** [review]

File: log/consul_test.go [hunk 1]
```diff
@@ -52,6 +52,36 @@ func TestConsul(t *testing.T) {
...
+		t.Log("Set level - NOOP")
+		buf.Reset()
+	})
+
+	t.Run("Log", func(t *testing.T) {
+		logger.Log(hclog.NoLevel, "123")
+		assert.Empty(t, buf.String())
+		buf.Reset()
+		logger.Log(hclog.Trace, "123")
+		assert.Contains(t, buf.String(), "DEBUG: 123")
```
Comment (Lazaros-Tsigkakos) [inline log/consul_test.go:66]: Should be `"TRACE: 123"` instead.

**Review by Lazaros-Tsigkakos — APPROVED** [review]
Nice addition indeed. LGTM, Sotiris.

File: log/consul_test.go (hunk 1 above)
Comment (mantzas) [reply log/consul_test.go:66]: This is the consul logger and funny enough trace reports `DEBUG`. The test is passing.

**Pull request merged**